src/csvcubed/models/cube/qb/components/unit.py

```diff
@@ -41,51 +64,64 @@ class NewQbUnit(QbUnit, UriIdentifiable, ArbitraryRdf):
     uri_safe_identifier_override: Optional[str] = field(default=None, repr=False)
     arbitrary_rdf: List[TripleFragmentBase] = field(default_factory=list, repr=False)
-
     base_unit: Optional[QbUnit] = field(default=None, repr=False)
     """
     The unit that this new unit is based on.
-    
+
     Codependent with :attr:`base_unit_scaling_factor`.
     """
     base_unit_scaling_factor: Optional[float] = field(default=None, repr=False)
     """
     How to scale the value associated with this unit to map back to the base unit.
-    
+
     Codependent with :attr:`base_unit`.
-    
-    e.g. if the base unit is *meters* and this unit (*kilometers*) has a scaling factor of **1,000**, then you multiply 
-    the value in *kilometers* by **1,000** to get the value in *meters*.    
-    """
 
+    e.g. if the base unit is *meters* and this unit (*kilometers*) has a scaling factor of **1,000**, then you multiply
+    the value in *kilometers* by **1,000** to get the value in *meters*.
+    """
     qudt_quantity_kind_uri: Optional[str] = field(default=None, repr=False)
     """
     The URI of the `qudt:QuantityKind` family to which this unit belongs.
-    
-    Codependent with :attr:`si_base_unit_conversion_multiplier`. 
+
+    Codependent with :attr:`si_base_unit_conversion_multiplier`.
     """
     si_base_unit_conversion_multiplier: Optional[float] = field(
         default=None, repr=False
     )
     """
     Multiply a value by this number to convert between this unit and its corresponding **SI unit**.
 
-    Codependent with :attr:`qudt_quantity_kind_uri`. 
-       
+    Codependent with :attr:`qudt_quantity_kind_uri`.
+
     See https://github.com/qudt/qudt-public-repo/wiki/User-Guide-for-QUDT#4-conversion-multipliers-in-qudt to understand
     the role of `qudt:conversionMultiplier` before using this. *It may not represent what you think it does.*
-    
-    SI - https://en.wikipedia.org/wiki/International_System_of_Units 
+
+    SI - https://en.wikipedia.org/wiki/International_System_of_Units
     """
 
-    optional_attribute_dependencies = enforce_optional_attribute_dependencies(
+    def _get_validations(self) -> Dict[str, ValidationFunction]:
+        return {
+            "label": validate_str_type,
+            "description": validate_optional(validate_str_type),
+            "source_uri": validate_optional(validate_uri),
+            **UriIdentifiable._get_validations(self),
+            "arbitrary_rdf": validate_list(validate_triple_fragment),
+            "base_unit": validate_optional(v.validated_model(QbUnit)),
+            "base_unit_scaling_factor": validate_optional(validate_float_type),
+            "qudt_quantity_kind_uri": validate_optional(validate_str_type),
+            "si_base_unit_conversion_multiplier": validate_optional(
+                validate_float_type
+            ),
+        }
+
+    optional_attribute_dependencies = pydantic_enforce_optional_attribute_dependencies(
```

Comment:
We need an equivalent function for this pydantic validator in the new `ValidatedModel` world.

Reply:
Done (thanks!)